aws-android-sdk-iot/src/main/java/com/amazonaws/mobileconnectors/iot/AWSIotMqttManager.java

```diff
@@ -730,6 +774,7 @@ private void initDefaults() {
         connectionStabilityTime = DEFAULT_CONNECTION_STABILITY_TIME_SECONDS;
         unitTestMillisOverride = null;
         needResubscribe = true;
+        this.customWebsocketHeaders = new Properties();
```

Comment:
Ordinarily `java.util.Properties` is used to read a `something.properties` file, which contains a list of key-value pairs, separated by `=`. Java `Properties` has been part of the Java ecosystem since 1.0, but is not widely used, today.

For example, a Java properties file looks like:
```
# config.properties
post=8888
host=coolhost.net
```

Android largely does not use these files, or this utility class. I was trying to think _why_ that is, and I'm not entirely sure. But here are two theories.

1. Android included a rich notion of `Resources`, which is more feature-rich and capable than the `Properties`.
2. Android is a younger derivative of Java. Android 1 was built on Java 1.5, and even by Gingerbread (2.3), Android was at 1.6. Point in short: the original practices of Java 1.0 were more distant in memory than the original Android ecosystem.

So anyway - TLDR, I'd use a `Map<String, String>` here, actually. `Properties` models a properties file, and wraps a `Map`. these headers here are literally a dictionary of key/values, without any other interesting behaviors.

Reply:
I agree but Paho uses `java.util.Properties` for all of their key/value storage needs.
https://github.com/eclipse/paho.mqtt.java/blob/master/org.eclipse.paho.client.mqttv3/src/main/java/org/eclipse/paho/client/mqttv3/MqttConnectOptions.java#L76

Reply:
Got it, makes sense!